docs/src/components/MainSectionSubsection.js

```diff
@@ -32,7 +31,7 @@ const MainSectionSubsection = ({ children, columns = 1, description, title }: Pr
             data-anchor
           >
             <Flex alignItems="center" gap={2}>
-              <Heading size="sm">{capitalizeFirstLetter(title)}</Heading>
+              <Heading size="sm">{title}</Heading>
```

Comment:
@ayeshakmaz Instead of doing this `capitalizeFirstLetter`, I would recommend to fix the issues instead. 

In our case we want to show
```
gestalt/button-icon-restrictions
```

and not

```
Gestalt/button-icon-restrictions
```

as the heading